### PR TITLE
Add support for 2.5D geometries to memory provider

### PR DESCRIPTION
**Pull request by alexbruy**
Also fixes related Processing issue

### CODE REVIEW

File: src/providers/memory/qgsmemoryprovider.cpp [hunk 1]
```diff
@@ -60,6 +60,18 @@ QgsMemoryProvider::QgsMemoryProvider( const QString& uri )
     mWkbType = QgsWkbTypes::MultiLineString;
   else if ( geometry == "multipolygon" )
     mWkbType = QgsWkbTypes::MultiPolygon;
+  else if ( geometry == "point25d" )
```
Comment (nyalldawson) [inline src/providers/memory/qgsmemoryprovider.cpp]: this block should use QgsWkbTypes::parseType() - then we'd also get correct handling of Z/M/curved (+less duplicate code)

File: src/providers/memory/qgsmemoryprovider.cpp [hunk 2]
```diff
@@ -229,6 +241,24 @@ QString QgsMemoryProvider::dataSourceUri( bool expandAuthConfig ) const
     case QgsWkbTypes::MultiPolygon :
       geometry = "MultiPolygon";
       break;
+    case QgsWkbTypes::Point25D :
```
Comment (nyalldawson) [inline src/providers/memory/qgsmemoryprovider.cpp]: ...and this should use QgsWkbTypes::displayString() for the same reason

**Comment by nyalldawson** [comment]
Looks good apart from the comments above. test_provider_memory.py testLayerGeometry() also needs to be updated to handle the new geometry types.

**Comment by alexbruy** [comment]
@nyalldawson, @m-kuhn thanks for your feedback. Comments addressed and hope this can be merged now.

File: python/plugins/processing/tools/vector.py [hunk 3]
```diff
@@ -60,6 +60,12 @@
     QgsWkbTypes.MultiPoint: 'MultiPoint',
     QgsWkbTypes.MultiLineString: 'MultiLineString',
     QgsWkbTypes.MultiPolygon: 'MultiPolygon',
+    QgsWkbTypes.Point25D: 'Point25D',
+    QgsWkbTypes.LineString25D: 'LineString25D',
+    QgsWkbTypes.Polygon25D: 'Polygon25D',
+    QgsWkbTypes.MultiPoint25D: 'MultiPoint25D',
+    QgsWkbTypes.MultiLineString25D: 'MultiLineString25D',
+    QgsWkbTypes.MultiPolygon25D: 'MultiPolygon25D',
 }
```
Comment (nyalldawson) [inline python/plugins/processing/tools/vector.py]: Everything else looks good - but I wonder about using GEOM_TYPE_MAP in the creation of memory layers (eg line 570). Doesn't this break support for M/Z/ZM geometry types? That line (570) could be replaced again by using QgsWkbTypes.displayString() and then it's both future proof and also would handle all geometry types.

**Pull request merged**